specs/Menu.md

```diff
@@ -1,731 +1,540 @@
 
-Since the SubmenuItem is effectively the same as the MenuItem, it should have only one slot for the Root `div` and for all intents and purposes should behave exactly the same as the MenuItem slot does.
+`MenuList` can also be used separately as the standalone variant of the `Menu`, since it should not control popup positioning or triggers. It is the only component in the API that can be used standalone. Envisioned to be used with more complex popup or trigger scenarios where the `Menu` component does not provide enough control for these situations.
 
-### Behaviors
+### MenuSection
 
-#### Menu
+Creates a section inside a `MenuList`, setting up header layout and dividers between `MenuItems`.
 
-The menu itself should be a list that renders menu items, it will provide context about its overall state but should not pass that directly into its items as props.
+The MenuSection is also a useful component to declare different selection groups (checkbox/radio) in a `MenuList`.
 
-##### Orientation
+### MenuItem
 
-The menu should provide some help for orienting its contents either vertically, like a left nav, or horizontally, like a nav bar. Ideally this will just be a prop that gets put into context so each item can decide how it should appear. It's possible that the menu should enforce direction itself.
+As the name infers
 
-##### Focus
+### SubMenu
 
-By default Focus should go to the tab stops and it's up to each menuItem to decide if it has tabindex=0. An example of a menuItem that shouldn't be focusable would be a Divider.
+Creates a `Menu` component with `MenuItem` trigger and handles the positioning of the nested menu.
 
-However for the menu we should provide a focus wrapper that is able to track focus using role="menuitem" to determine what item should get focused.
+## Sample code
 
-The menu does not need to provide a native way to be controlled without getting focus directly. Like above, there should be a simple utility wrapper that could be written to implement this.
+The below samples do not represent the definitive props of the final implemented component, but represent the ideal final implementations. Can be subject to change during the implementation phase.
 
-##### On Menu Click
+### Default Menu
 
-Some implementations of menu have a general `onClick` function that is applied to all items as a way of getting more information.
+```typescript
+const trigger = <button> Open menu </button>
 
-Recommendation: There is not a need to have an `onClick` that is called whenever an item is clicked. Instead the root `menu` should take an `onClick` that will fire if a child's on click does not prevent default.
+const menu = (
+  <Menu trigger={trigger}>
+    <MenuItem title="Option 1" />
+    <MenuItem title="Option 2" />
+    <MenuItem title="Option 3" />
+  <Menu>
+)
+```
 
-#### Menu Items
+### Sections and submenus
+
+```typescript
+const trigger = <button> Open menu </button>
+
+const menu = (
+  <Menu trigger={trigger}>
```

Comment:
In the prototype we intentionally skipped `trigger` as it breaks JSX layout:

```jsx
<Menu>
  <MenuItem>
  {/* nested item ⬇️ */}
  <Menu trigger={<MenuItem />> {/* what will be rendered in the layout ❔ */}
    <MenuItem />
  </>
</Menu>
```

Reply:
My suggestion was to have a separate `SubMenu` component in these situations. The root level `Menu` for me only exists as sugar for positioning

Reply:
Have you considered some kind of "MenuAnchor" wrapper element that wraps the trigger and has a slot for the menu? This is just brainstorming, but perhaps something like:
```tsx
<MenuAnchor menu={
    <Menu>
      <MenuItem>Item 1</MenuItem>
      <MenuItem>Item 2</MenuItem>
    </Menu>}>
  <Button>Open Menu</Button>
</MenuAnchor>
```

That might also be useful for separating the responsibility for creating and positioning the popup (handled by MenuAnchor), and rendering the content of the menu (handled by Menu/MenuItem). That way a menu could be rendered inside another container that's not a popup (or a custom popup).

Reply:
@behowell  I would avoid wrapping the trigger and putting the menu in a slot. This is because I would like `Menu` to reflect correct dom structure with JSX as much as possible to have an API consistent with all the other menu implementations in react that are out there.

Please feel free to join the trigger deep dive and weigh in with your ideas